test/framework/src/main/java/org/elasticsearch/test/transport/MockTransportService.java

```diff
@@ -437,18 +441,26 @@ public void sendRequest(
                 String action,
                 TransportRequest request,
                 TransportRequestOptions options
-            ) {
-                // don't send anything, the receiving node is unresponsive
-                toClose.add(connection);
+            ) throws IOException {
+                if (connection.isClosed()) {
+                    throw new NodeNotConnectedException(connection.getNode(), "connection already closed");
+                } else if (sendSemaphore.tryAcquire()) {
+                    // don't send anything, the receiving node is unresponsive
+                    toClose.add(connection);
+                    sendSemaphore.release();
+                } else {
+                    connection.sendRequest(requestId, action, request, options);
```

Comment:
Unsure about this, should we close the connection immediately here perhaps?

Reply:
I don't think so? We hand the real connection in so that we can use it if necessary. If the user of this class has cleared the disruption behavior, I think the correct behavior is to revert back to normal instead of continuing to disrupt. This will rarely happen as this is explicitly a race.

Reply:
Ok yes there's a race here, in the middle of removing the disruption the connection may be closed or may be used to send the request so really it's valid to choose either.